src/core/Internal.ml

```diff
@@ -1480,7 +1484,7 @@ module Make(Plugin : PLUGIN)
           enqueue_bool st fuip ~level:cr.cr_backtrack_lvl (Bcp lclause)
         ) else (
           assert Plugin.mcsat;
-          st.next_decision <- Some fuip.neg
```

Comment:
We should probably add an assertion that the `next_decisions` list is empty here I think.